### PR TITLE
Switch PyTorch PTQ to common implementation

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

- Switch `nncf.quantize` to post training implementation for PT backend
- Removed OLD_TORCH backend from test_quantize_conformance.py
- Add support of dict as output of calibration dataset

### Related tickets

119910

### CODE REVIEW

**Comment by AlexanderDokuchaev** [comment]
post_training_quantization/193/

File: nncf/torch/engine.py [hunk 1]
```diff
@@ -32,12 +32,17 @@ def __init__(self, model: nn.Module):
...
         Runs Torch model on the provided input.
 
         :param input_data: inputs for the model
         :return output_data: model outputs
         """
-
+        if isinstance(input_data, dict):
+            return self._model(**input_data)
+        if isinstance(input_data, tuple):
+            return self._model(*input_data)
```
Comment (vshampor) [inline nncf/torch/engine.py:47]: Which tests caught this?

Comment (AlexanderDokuchaev) [reply nncf/torch/engine.py:47]: 'tests/torch/test_transform_fn.py'
But there was missed `dict` as input, added it.

Comment (AlexanderDokuchaev) [reply nncf/torch/engine.py:47]: Looks like this changes conflicts with https://github.com/openvinotoolkit/nncf/pull/2196, dict and tuple will supported after your PR?

**Review by alexsu52 — APPROVED** [review]
LGTM

File: nncf/torch/engine.py [hunk 2]
```diff
@@ -32,12 +32,17 @@ def __init__(self, model: nn.Module):
...
-    def infer(self, input_data: Union[torch.Tensor, Dict[str, torch.Tensor]]) -> Union[torch.Tensor, Dict[str, Any]]:
+    def infer(
+        self, input_data: Union[torch.Tensor, Tuple[torch.Tensor], Dict[str, torch.Tensor]]
+    ) -> Union[torch.Tensor, Dict[str, Any]]:
         """
         Runs Torch model on the provided input.
 
         :param input_data: inputs for the model
         :return output_data: model outputs
         """
```
Comment (daniil-lyakhov) [inline nncf/torch/engine.py:43]:
```suggestion
        :param input_data: Inputs for the model.
        :return: Model outputs.
        """
```

File: nncf/torch/quantization/quantize_model.py [hunk 3]
```diff
@@ -9,190 +9,84 @@
...
+def create_nncf_network(model: torch.nn.Module, dataset: Dataset) -> NNCFNetwork:
     """
-    This class wraps the nncf.Dataset.
+    Creates NNCFNetwork instance for the PyTorch model where the first item of dataset
+    is used for model tracing.
 
-    This is required for proper initialization of certain compression algorithms.
+    :param model: PyTorch model
+    :param dataset: Dataset for model tracing
+    :return: NNCFNetwork instance for the input model
```
Comment (daniil-lyakhov) [inline nncf/torch/quantization/quantize_model.py:44]:
```suggestion
    :param model: PyTorch model.
    :param dataset: Dataset for model tracing.
    :return: NNCFNetwork instance for the input model.
```

**Review by daniil-lyakhov — APPROVED** [review]
Minor

**Comment by MaximProshin** [comment]
@vshampor , please state explicitly if there are any remaining comments from you that should be applied or you're ok to merge it.

File: nncf/torch/engine.py (hunk 1 above)
Comment (vshampor) [reply nncf/torch/engine.py:47]: I guess we'll have to see, based on your tests.

**Comment by vshampor** [comment]
I'm not blocking it, but the change will have to be aligned with #2196 in one way or another.

**Pull request merged**